primitives/container-chain-genesis-data/src/lib.rs

```diff
@@ -68,18 +74,53 @@ pub mod json;
 )]
 #[serde(bound = "")]
 pub struct ContainerChainGenesisData {
-    pub storage: Vec<ContainerChainGenesisDataItem>,
-    // TODO: make all these Vec<u8> bounded
-    #[serde(with = "sp_core::bytes")]
-    pub name: Vec<u8>,
-    #[serde(with = "sp_core::bytes")]
-    pub id: Vec<u8>,
-    pub fork_id: Option<Vec<u8>>,
-    #[serde(with = "sp_core::bytes")]
-    pub extensions: Vec<u8>,
+    pub storage: BoundedVec<ContainerChainGenesisDataItem, ConstU32<655350>>,
```

Comment:
What is this number? The old limit was 5MB

Reply:
The capacity of `Vec` calculations:
- The limit still - 5MB
- In case we have the storage with an empty key/value, it will take 4bytes + 4bytes = 8 bytes for one record
- 5MB / 8 bytes = 655360 empty items.
So, the maximum amount of empty records can be up to 655360.

Reply:
That size is not correct but it doesn't matter, the limit should be high enough.

This test prints 2002 as the size, I guess 2 bytes to encode that there are 1000 elements, and then 2 bytes per element (1 byte to encode 0 len key and 1 byte to encode 0 len value).

```rust
let is = ContainerChainGenesisDataItem {
    key: vec![],
    value: vec![],
};

let mis = vec![is; 1000];

println!("{}", mis.encoded_size());
```

But keys will never be empty so that's not realistic, and in the end the 5MB limit will be hit before this one in practice. So not sure what to put here, I would just leave it unbounded but that's not possible.

In any case, add a short comment explaining how to derive the limit, like

// Assume 5MB max total size / 8 bytes per item = 655360 items

Reply:
Updated the comment